### PR TITLE
Add AzureStack Hub Admin VPN API

### PR DESCRIPTION
**Pull request by iancaragol**
This is a new read-only API for Azure Stack Administrators. It lists all VPN resources that have been created on the stamp and enables the administrator to better manage their limited VPN capacity.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  
  Goal is to deploy the new api in the first Azure Stack release of next year. Probably towards late Jan/early Feb
  
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  
  late Jan/early Feb
  
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by iancaragol** [comment]
[Overview & Documentation](https://microsoft.sharepoint.com/teams/AzureStack/_layouts/OneNote.aspx?id=%2Fteams%2FAzureStack%2FAzure%20Stack%20Shared%2FNetworking%2FAzure%20Stack%20NET&wd=target%28Nickel.one%7CF9632A0F-0368-49CD-A79D-FF6796C10CFD%2FOverview%7C2E309CA8-547D-44E1-B087-3D0A509E4E37%2F%29)

This API + new UI will enable Azure Stack Administrators to see all VPN resources that have been created on the stamp. Using this information they will be able to:

- Identify useless or disconnected VPN tunnels that are consuming capacity
- Work with tenants to manage the stamp's VPN capacity
- Work with CSS to more quickly and easily diagnose VPN issues

Targeting the first Azure Stack release of 2022, probably in late January/early February 

Example UI Mockups:

image

image

**Review by bganapa — APPROVED** [review]

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          "default": {
+            "description": "Error Response.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:52]: This should be
"nextLinkName": "nextLink"

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      }
+    }
+  },
+  "definitions": {
+    "VirtualNetworkGatewayConnectionsList": {
+      "description": "List of Virtual Network Gateway Connections.",
+      "type": "object",
+      "properties": {
+        "value": {
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:61]: >value [](http://example.com/codeflow?start=9&length=5)

nit: declare this as readonly

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+          "description": "List of Virtual Network Gateway Connections.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VirtualNetworkGatewayConnection"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:69]: >nextLink [](http://example.com/codeflow?start=9&length=8)

nit: declare this as readonly

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+            "modelAsString": true,
+            "name": "VirtualNetworkGatewayConnectionConnectionStateType"
+          }
+        },
+        "activeGateway": {
+          "description": "Active Gateway VM that the connection is configured on.",
+          "type": "string",
+          "readOnly": true
+        },
+        "connectionError": {
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:192]: >connectionError [](http://example.com/codeflow?start=9&length=15)

Sounds like this is a readonly property. If yes, define it as readonly.

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    },
+    "VirtualNetworkGatewayConnectionProperties": {
+      "description": "Virtual Network Gateway Connection properties.",
+      "type": "object",
+      "properties": {
+        "connectionState": {
+          "description": "Virtual Network Gateway Connection State.",
+          "$ref": "#/definitions/VirtualNetworkGatewayConnectionState"
+        },
+        "provisioningState": {
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:104]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

Declare this as enum property specifying the allowed values.

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network.Admin/adminVirtualNetworkGatewayConnections": {
+      "get": {
```
Comment (RamyasreeChakka) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:20]: In addition to the list API, please define & implement below API for GET on individual resource. 

GET /subscriptions/{subscriptionId}/providers/Microsoft.Network.Admin/adminVirtualNetworkGatewayConnections/{connectionName}

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json (hunk 1 above)
Comment (iancaragol) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:52]: Addressed, thanks!

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json (hunk 3 above)
Comment (iancaragol) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:70]: Addressed, thanks!

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json (hunk 5 above)
Comment (iancaragol) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/stable/2022-02-01/VirtualNetworkGatewayConnections.json:106]: Addressed, thanks!

**Review by RamyasreeChakka — APPROVED** [review]
:shipit:

**Comment by RamyasreeChakka** [comment]
@iancaragol Signed off from ARM side.

**Pull request merged**